server/src/main/java/io/crate/types/DataType.java

```diff
@@ -105,6 +105,18 @@ public T explicitCast(Object value) throws IllegalArgumentException, ClassCastEx
         return implicitCast(value);
     }
 
+    /**
+     * To prepare a value of the same {@link DataType<T>} for insertion.
+     * The method fall backs to the implicit cast if it is overwritten
+     * by a data type subclass.
+     *
+     * @param value The value of the {@link DataType<T>}.
+     * @return The prepared for insertion value of the {@link DataType<T>}.
+     */
+    public T valueForInsert(Object value) {
+        return implicitCast(value);
```

Comment:
I think the default implementation should return the value as-is unchanged. Has a lower overhead and is closer to the prior behavior before the introduction of the method.

Will also be easier to reason about what is happening.